### ISSUE TITLE
[Bug]: PHP8.1 - Error: trim(): Passing null to parameter #1 ($string) of type string is deprecated at /apps/sharebymail/lib/Activity.php#323

### ISSUE THREAD

**Issue by solracsf** [nextcloud/server#35168]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

See error log.

### Steps to reproduce

N/A

### Expected behavior

No error.

### PHP engine version

PHP 8.1

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
N/A
```

### List of activated Apps

```shell
N/A
```

### Nextcloud Logs

```shell
{
  "reqId": "ka3qVmvHEtZVtjDLkPC7",
  "level": 3,
  "time": "2022-11-15T08:10:55+00:00",
  "remoteAddr": "90.119.179.128",
  "app": "PHP",
  "method": "GET",
  "url": "/ocs/v2.php/apps/activity/api/v2/activity/filter?format=json&object_type=files&object_id=61265409",
  "message": "trim(): Passing null to parameter #1 ($string) of type string is deprecated at /var/www/nextcloud/apps/sharebymail/lib/Activity.php#323",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36",
  "version": "24.0.7.1",
  "exception": {
    "Exception": "Error",
    "Message": "trim(): Passing null to parameter #1 ($string) of type string is deprecated at /var/www/nextcloud/apps/sharebymail/lib/Activity.php#323",
    "Code": 0,
    "Trace": [
      {
        "function": "onError",
        "class": "OC\\Log\\ErrorHandler",
        "type": "::",
        "args": [
          8192,
          "trim(): Passing null to parameter #1 ($string) of type string is deprecated",
          "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
          323
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 323,
        "function": "trim",
        "args": [
          null,
          "/"
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 290,
        "function": "generateFileParameter",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          61265409,
          null
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 118,
        "function": "getParsedParameters",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          {
            "__class__": "OC\\Activity\\Event"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 102,
        "function": "parseShortVersion",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          {
            "__class__": "OC\\Activity\\Event"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/GroupHelper.php",
        "line": 86,
        "function": "parse",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          "fr",
          {
            "__class__": "OC\\Activity\\Event"
          },
          {
            "__class__": "OC\\Activity\\Event"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Data.php",
        "line": 255,
        "function": "addActivity",
        "class": "OCA\\Activity\\GroupHelper",
        "type": "->",
        "args": [
          [
            29856848,
            1667039402,
            30,
            "shared",
            "",
            "And 10 more entries, set log level to debug to see all entries"
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Controller/APIv2Controller.php",
        "line": 273,
        "function": "get",
        "class": "OCA\\Activity\\Data",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Activity\\GroupHelper"
          },
          {
            "__class__": "OCA\\Activity\\UserSettings"
          },
          "***********",
          0,
          49,
          "desc",
          "filter",
          "files",
          61265409
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Controller/APIv2Controller.php",
        "line": 209,
        "function": "get",
        "class": "OCA\\Activity\\Controller\\APIv2Controller",
        "type": "->",
        "args": [
          "filter",
          0,
          50,
          false,
          "files",
          61265409,
          "desc"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 225,
        "function": "getFilter",
        "class": "OCA\\Activity\\Controller\\APIv2Controller",
        "type": "->",
        "args": [
          "filter",
          0,
          50,
          false,
          "files",
          61265409,
          "desc"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 133,
        "function": "executeController",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Activity\\Controller\\APIv2Controller"
          },
          "getFilter"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/App.php",
        "line": 172,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Activity\\Controller\\APIv2Controller"
          },
          "getFilter"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/Route/Router.php",
        "line": 298,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\Activity\\Controller\\APIv2Controller",
          "getFilter",
          {
            "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
          },
          [
            "filter",
            "ocs.activity.APIv2.getFilter"
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/ocs/v1.php",
        "line": 62,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->",
        "args": [
          "/ocsapp/apps/activity/api/v2/activity/filter"
        ]
      },
      {
        "file": "/var/www/nextcloud/ocs/v2.php",
        "line": 23,
        "args": [
          "/var/www/nextcloud/ocs/v1.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/nextcloud/lib/private/Log/ErrorHandler.php",
    "Line": 92,
    "CustomMessage": "--"
  },
  "id": "63734a0e0f96d"
}
```

**Comment by szaimen** [comment]
cc @come-nc

**Comment by come-nc** [comment]
@solracsf Any tips on how to reproduce that?
The trace implies an activity event is triggered on a file with a `null` path. Is there anything special about the file you share by mail? Or was it deleted?

**Comment by solracsf** [comment]
@come-nc how can i get the file name/path on the filesystem? Otherwise how can i check if file even exists?
FYI, these are immediately followed by:

```
{
  "reqId": "juoEWNnEZPbcZDLKahAL",
  "level": 3,
  "time": "2022-12-16T19:00:10+00:00",
  "app": "PHP",
  "method": "GET",
  "url": "/ocs/v2.php/apps/activity/api/v2/activity/filter?format=json&object_type=files&object_id=61912508",
  "message": "basename(): Passing null to parameter #1 ($path) of type string is deprecated at /var/www/nextcloud/apps/sharebymail/lib/Activity.php#322",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36",
  "version": "24.0.8.2",
  "exception": {
    "Exception": "Error",
    "Message": "basename(): Passing null to parameter #1 ($path) of type string is deprecated at /var/www/nextcloud/apps/sharebymail/lib/Activity.php#322",
    "Code": 0,
    "Trace": [
      {
        "function": "onError",
        "class": "OC\\Log\\ErrorHandler",
        "type": "::",
        "args": [
          8192,
          "basename(): Passing null to parameter #1 ($path) of type string is deprecated",
          "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
          322
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 322,
        "function": "basename",
        "args": [
          null
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 290,
        "function": "generateFileParameter",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          61912508,
          null
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 118,
        "function": "getParsedParameters",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          {
            "__class__": "OC\\Activity\\Event"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/sharebymail/lib/Activity.php",
        "line": 102,
        "function": "parseShortVersion",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          {
            "__class__": "OC\\Activity\\Event"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/GroupHelper.php",
        "line": 86,
        "function": "parse",
        "class": "OCA\\ShareByMail\\Activity",
        "type": "->",
        "args": [
          "fr",
          {
            "__class__": "OC\\Activity\\Event"
          },
          {
            "__class__": "OC\\Activity\\Event"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Data.php",
        "line": 255,
        "function": "addActivity",
        "class": "OCA\\Activity\\GroupHelper",
        "type": "->",
        "args": [
          [
            31194235,
            1671217072,
            30,
            "shared",
            "marc.verrier@alixiomobilite.fr",
            "And 10 more entries, set log level to debug to see all entries"
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Controller/APIv2Controller.php",
        "line": 273,
        "function": "get",
        "class": "OCA\\Activity\\Data",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Activity\\GroupHelper"
          },
          {
            "__class__": "OCA\\Activity\\UserSettings"
          },
          "marc.verrier@alixiomobilite.fr",
          0,
          47,
          "desc",
          "filter",
          "files",
          61912508
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Controller/APIv2Controller.php",
        "line": 209,
        "function": "get",
        "class": "OCA\\Activity\\Controller\\APIv2Controller",
        "type": "->",
        "args": [
          "filter",
          0,
          50,
          false,
          "files",
          61912508,
          "desc"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 225,
        "function": "getFilter",
        "class": "OCA\\Activity\\Controller\\APIv2Controller",
        "type": "->",
        "args": [
          "filter",
          0,
          50,
          false,
          "files",
          61912508,
          "desc"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 133,
        "function": "executeController",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Activity\\Controller\\APIv2Controller"
          },
          "getFilter"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/App.php",
        "line": 172,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Activity\\Controller\\APIv2Controller"
          },
          "getFilter"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/Route/Router.php",
        "line": 298,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\Activity\\Controller\\APIv2Controller",
          "getFilter",
          {
            "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
          },
          [
            "filter",
            "ocs.activity.APIv2.getFilter"
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/ocs/v1.php",
        "line": 62,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->",
        "args": [
          "/ocsapp/apps/activity/api/v2/activity/filter"
        ]
      },
      {
        "file": "/var/www/nextcloud/ocs/v2.php",
        "line": 23,
        "args": [
          "/var/www/nextcloud/ocs/v1.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/nextcloud/lib/private/Log/ErrorHandler.php",
    "Line": 92,
    "CustomMessage": "--"
  },
  "id": "639db1e0eecf7"
}
```

**Comment by szaimen** [comment]
Hi, please update to 25.0.7 or better 26.0.2 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 26-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by solracsf** [comment]
Not fixed on 25.0.7 RC1

**Comment by come-nc** [comment]
> Not fixed on 25.0.7 RC1

Are you sure it is not triggered by activity events from before you updated to 25.0.7, which are still in database?

**Comment by solracsf** [comment]
Well i can't see these in the logs of last 2 months, closing for now.
